pilot/pkg/model/context.go

```diff
@@ -300,6 +300,9 @@ type Proxy struct {
 	// WatchedResources contains the list of watched resources for the proxy, keyed by the DiscoveryRequest TypeUrl.
 	WatchedResources map[string]*WatchedResource
 
+	// WatchedResourcesList contains the ordered list of watched resources for the proxy, ordered in accordance with known push order.
+	WatchedResourcesList []*WatchedResource
```

Comment:
I donot like adding a new confusing field to Proxy, this struct is already very complex

Reply:
What is confusing here? Do you want to order it every time? If that is the preference I can change it back. @howardjohn WDYT?

Reply:
Since you mentioned this is confusing, I just reverted this part of the change and doing ordering every time as we were doing earlier. PTAL.